compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -615,6 +615,16 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
         .flatten();
     features.extend(feats);
 
+    // -Zfixed-x18
+    if sess.opts.unstable_opts.fixed_x18 {
+        if sess.target.arch != "aarch64" {
+            // TODO: What's the correct way to return a error here?
+            panic!("-Zfixed-x18 only allowed on aarch64");
+        }
```

Comment:
Would anyone be able to point me in the right direction here? Will I need to add a new type for this error? Which file?

Also, where should the test for this failure go?

Reply:
(Repeating what I said on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/New.20.60reserve-x18.60.20target.20feature/near/438638431) in case discussion happens here instead)

Take a look at https://github.com/rust-lang/rust/blob/518becf5ea30807f98b89002a70e5640e6e8bbf4/compiler/rustc_codegen_llvm/src/errors.rs to see the error structs. They're emitted from the diagnostics context, e.g. with `sess.dcx().emit_err()`. [This chapter](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-structs.html) of the dev guide also has more information.

Reply:
Thanks! The PR has been updated with a proper error for this.